src/vs/vscode.d.ts

```diff
@@ -5194,6 +5194,14 @@ declare module 'vscode' {
 		export let workspaceFolders: WorkspaceFolder[] | undefined;
 
 		/**
+		 * The name of the workspace. `undefined` when no folder
+		 * has been opened.
+		 *
+		 * @readonly
+		 */
+		export let name: string | undefined;
```

Comment:
Can this change and iff so would we want an event for this?

Reply:
@jrieken currently the name cannot change because we do not expose a name property in the workspace config file yet. However there is an issue for it and once we add that we should wire this in properly.